test/test_utilities.cpp

```diff
@@ -107,14 +90,49 @@ void ValidateTensorData(Tensor& T1, Tensor& T2, float tol) {
   }
 }
 
+// Compares Tensor vectors
+void Compare(const vector<Tensor>& v1, const vector<Tensor>& v2) {
+  ASSERT_EQ(v1.size(), v2.size());
+  for (int i = 0; i < v1.size(); i++) {
+    auto expected_dtype = v1[i].dtype();
+    switch (expected_dtype) {
+      case DT_FLOAT:
+        Compare<float>(v1[i], v2[i]);
+        break;
+      case DT_INT8:
+        Compare<int8>(v1[i], v2[i]);
+        break;
```

Comment:
Slight confusion here:
Compare is an untemplated function that compares 2 vectors of tensors

Compare is also templated function that compares individual tensors?

Maybe we should have different and more descriptive names for them? especially because in a situation when someone wants to compare 2 tensors, one should do this
```
Compare({t1}, {t1}) . //pass singleton vectors
```

But by mistake if you write:
```
Compare(t1, t2)
```

Then the compiler will understand it as 
```
template <class T>
bool Compare(T arg0, T arg1) {
  return arg0 == arg1;
}
```

which was not our intention




Compare is also overloaded to compare POD:
```Compare<int8>(v1[i], v2[i]);```

Reply:
There is a Compare function for tensors too, so when someone calls 
```
Compare(t1, t2)
```
It calls
```
// Compares two Tensors
template <typename T>
void Compare(const Tensor& T1, const Tensor& T2) {
```
and not the
```
// Compares two arguments
template <class T>
bool Compare(T arg0, T arg1) {
  return arg0 == arg1;
}
```
I tested this with logs. So, we might be safe here.

However, I see this being a problem for unsupported types.

For eg some new class A
```
Compare(a1, a2) 
```
might get us into trouble.

Are there any other corner cases where the Compare function overload might be an issue?